File: archived/tests/transpiler/x/php/cross_join_triple.php
<?php
$nums = [1, 2];
$letters = ["A", "B"];
$bools = [true, false];
$combos = [];
foreach ($nums as $n) {
  foreach ($letters as $l) {
    foreach ($bools as $b) {
      $combos[] = ["n" => $n, "l" => $l, "b" => $b];
    }
  }
}

echo "--- Cross Join of three lists ---", PHP_EOL;
foreach ($combos as $c) {
  echo (is_float($c["n"]) ? json_encode($c["n"], 1344) : $c["n"]) . " " . (is_float($c["l"]) ? json_encode($c["l"], 1344) : $c["l"]) . " " . (is_float($c["b"]) ? json_encode($c["b"], 1344) : $c["b"]), PHP_EOL;
}

File: archived/tests/transpiler/x/kt/cross_join_triple.kt
fun main() {
    val nums = mutableListOf(1, 2)
    val letters = mutableListOf("A", "B")
    val bools = mutableListOf(true, false)
    val combos = run {
    val _res = mutableListOf<MutableMap<String, Any>>()
    for (n in nums) {
        for (l in letters) {
            for (b in bools) {
                _res.add(mutableMapOf("n" to n, "l" to l, "b" to b))
            }
        }
    }
    _res
}
    println("--- Cross Join of three lists ---")
    for (c in combos) {
        println(listOf(c["n"], c["l"], c["b"]).joinToString(" "))
    }
}
